modules/engine/src/model/model.ts

```diff
@@ -348,10 +348,7 @@ export class Model {
 
     // Extract textures & framebuffers set by the modules
     // TODO better way to extract bindings
-    const keys = Object.keys(uniforms).filter(k => {
-      const uniform = uniforms[k];
-      return !isNumberArray(uniform) && (typeof uniform !== 'number') && (typeof uniform !== 'boolean');
-    });
+    const keys = Object.keys(uniforms).filter(k => !isUniform(uniforms[k]));
```

Comment:
Let's type out variable names.
```suggestion
    const keys = Object.keys(uniforms).filter(key => !isUniform(uniforms[key]));
```
Or better
```suggestion
    const uniformNames = Object.keys(uniforms).filter(name => !isUniform(uniforms[name]));
```

Reply:
Maybe a `splitUniformsAndBindings()` helper would help make code a bit leaner?